### DISCUSSION TITLE
Openwrt Cross Compile Support for Musl Mips32 after 3.1.2

### DISCUSSION THREAD

**Discussion by zloyle** [fish-shell/fish-shell#8638] [Q&A]
Has something major changed that prevents any version after 3.1.2 from working on a musl mipsel32 device?

Last working version I am able to successfully cross compile is 3.1.2. All versions afterword just hang and stop accepting input after the greeting forcing me to load another shell and kill the process to gain back control. Compiling latest master doesn't hang it just fails and thows this error.

Assertion failed: (status_.load() & STATUS_NEEDS_WAKEUP) == 0 && "No thread should be waiting" (/build/sources/fish/src/topic_monitor.cpp: try_update_gens_maybe_becoming_reader: 208)
Aborted

Is this to be expected? Thanks.

**Comment by krobelus** [comment]
There were some commits that touched the topic monitor; see `git log --grep=topic.monitor 3.1.2..3.2.0`
but the definitive way to find out is to bisect (I guess starting with those two versions and then later ones)
